### PR TITLE
Should throw IllegalStateException when transaction is not found and already exists

### PR DESCRIPTION
**Pull request by brfrn169**
Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/transaction/common/AbstractDistributedTransactionManager.java [hunk 1]
```diff
@@ -92,7 +92,8 @@ private void addActiveTransaction(DistributedTransaction transaction)
       throws TransactionException {
     if (activeTransactions.putIfAbsent(transaction.getId(), transaction) != null) {
       transaction.rollback();
-      throw new TransactionException("The transaction already exists");
+      throw new IllegalStateException(
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/common/AbstractDistributedTransactionManager.java:95]: Made it throw `IllegalStateException` instead of `TransactionException`.

File: core/src/main/java/com/scalar/db/transaction/common/AbstractDistributedTransactionManager.java [hunk 2]
```diff
@@ -101,14 +102,15 @@ private void removeActiveTransaction(String transactionId) {
...
 
   @Override
-  public DistributedTransaction resume(String txId) throws TransactionException {
+  public DistributedTransaction resume(String txId) {
     return activeTransactions
         .get(txId)
         .orElseThrow(
             () ->
-                new TransactionException(
+                new IllegalStateException(
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/common/AbstractDistributedTransactionManager.java:110]: Ditto.

File: core/src/main/java/com/scalar/db/transaction/common/AbstractTwoPhaseCommitTransactionManager.java [hunk 3]
```diff
@@ -93,7 +93,8 @@ public Optional<String> getTable() {
   private void addActiveTransaction(ActiveTransaction transaction) throws TransactionException {
     if (activeTransactions.putIfAbsent(transaction.getId(), transaction) != null) {
       transaction.rollback();
-      throw new TransactionException("The transaction already exists");
+      throw new IllegalStateException(
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/common/AbstractTwoPhaseCommitTransactionManager.java:96]: Ditto.

File: core/src/main/java/com/scalar/db/transaction/common/AbstractTwoPhaseCommitTransactionManager.java [hunk 4]
```diff
@@ -102,14 +103,15 @@ private void removeActiveTransaction(String transactionId) {
...
 
   @Override
-  public TwoPhaseCommitTransaction resume(String txId) throws TransactionException {
+  public TwoPhaseCommitTransaction resume(String txId) {
     return activeTransactions
         .get(txId)
         .orElseThrow(
             () ->
-                new TransactionException(
+                new IllegalStateException(
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/common/AbstractTwoPhaseCommitTransactionManager.java:111]: Ditto.

File: core/src/main/java/com/scalar/db/transaction/rpc/GrpcTransactionManager.java [hunk 5]
```diff
@@ -98,7 +98,7 @@ public DistributedTransaction begin(String txId) throws TransactionException {
   private DistributedTransaction beginInternal(@Nullable String txId) throws TransactionException {
     return executeWithRetries(
         () -> {
-          GrpcTransactionOnBidirectionalStream stream = getBidirectionalStream();
+          GrpcTransactionOnBidirectionalStream stream = getStream();
```
Comment (komamitsu) [inline core/src/main/java/com/scalar/db/transaction/rpc/GrpcTransactionManager.java:101]: This kind of changes doesn't seem related to this PR, but it's actually a part of this PR?

Comment (brfrn169) [reply core/src/main/java/com/scalar/db/transaction/rpc/GrpcTransactionManager.java:101]: It's not directly related to this PR actually. It's just renaming the private method for unit tests.

**Review by komamitsu — APPROVED** [review]
👍

**Comment by brfrn169** [comment]
Merging this for now.

@Torch3333 Can you please take a look when you come back?

**Review by Torch3333 — COMMENTED** [review]
LGTM, thank you!
